apm-agent-core/src/main/java/co/elastic/apm/agent/impl/ElasticApmTracer.java

```diff
@@ -317,6 +321,10 @@ public Span startSpan(AbstractSpan<?> parent, long epochMicros) {
         Transaction transaction = currentTransaction();
         if (transaction != null) {
             if (isTransactionSpanLimitReached(transaction)) {
+                if (lastSpanMaxWarningTimestamp == 0L || (epochMicros - lastSpanMaxWarningTimestamp > MAX_LOG_INTERVAL_MICRO_SECS)) {
+                    lastSpanMaxWarningTimestamp = epochMicros;
+                    logger.warn("Max spans ({}) for a transaction has been reached. For this transaction and possibly others, further spans will be dropped. See config param 'transaction_max_spans'.", coreConfiguration.getTransactionMaxSpans());
```

Comment:
The transaction name would be a useful hint even if it's only an example:
```suggestion
                    logger.warn("Max spans ({}) for transaction {} has been reached. For this transaction and possibly others, further spans will be dropped. See config param 'transaction_max_spans'.", coreConfiguration.getTransactionMaxSpans(), transaction.getName());
```

Reply:
Fixed in e910d86. `transaction.getName()` wasn't a thing and the `transaction.getNameAsString()` mentions only to use in test. So just defaulted to the toString() on transaction. I also updated the docs to call out the log statement is only after a period of time since adding the transaction in there made it feel like it needed called out.

Reply:
The usage warning is intended to avoid the String object allocation, but since this happens once every 5 minutes at most, this is not an issue. 
Let's merge it as is though.
Thanks a lot!